lib/portlayer/exec/container.go

```diff
@@ -161,7 +161,16 @@ func newContainer(base *containerBase) *Container {
 		// set state
 		switch base.Runtime.PowerState {
 		case types.VirtualMachinePowerStatePoweredOn:
-			c.state = StateRunning
+			// the containerVM is poweredOn, so set state to starting
+			// then check to see if a start was successful
+			c.state = StateStarting
+			// If any sessions successfully started then set to running
+			for _, s := range base.ExecConfig.Sessions {
+				if s.Started != "" {
+					c.state = StateRunning
```

Comment:
If we don't get to running here then it won't get  into a running state.  We'll need to figure out a way to handle those longer state transitions -- as noted here: https://github.com/vmware/vic/blob/master/lib/portlayer/exec/container.go#L349